plugin/helpers_bots.go

```diff
@@ -72,3 +105,44 @@ func (p *HelpersImpl) EnsureBot(bot *model.Bot) (retBotId string, retErr error)
 
 	return createdBot.UserId, nil
 }
+
+func (p *HelpersImpl) ShouldProcessMessage(post *model.Post, botUserId string, options ...ShouldProcessMessageOption) (bool, error) {
```

Comment:
Accepting a `botUserId` seems unexpected, since if I'm not using a bot but still want to exclude system messages or some such, I'd have to pass an empty string.

Would it make sense for this helper to lookup the bot id automatically, refactoring the code in `EnsureBot`? Then it would behave properly by default, and if the plugin wanted to exclude more user ids, they could do so manually?

Reply:
Yes it seems unexpected. So If I understand correctly I will have to use user id from the post ?
Otherwise how to lookup the bot automatically?

Reply:
@RajatVaryani, take a look at the `EnsureBot` implementation just above. Anyone using the helpers API will flow through here, and it uses a special key value entry to record the created bot.

Reply:
@lieut-data Thanks for that. I have made changes. Please review it.